File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        let totalP;
        let z = 0;
        for (const n of nums) {
            if (n == 0) {
                z++;
                continue;
            }
            totalP = (totalP || 1) * n;
        }
        const res = [];
        for (const n of nums) {
            if (n == 0) {
                if (z > 1) res.push(0);
                else res.push(totalP);
            } else {
                if (z > 0) res.push(0);
                else res.push(totalP / n);
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        l=0
        s=0
        while l < n:
            h=l+1
            while h < n:
                s=nums[l] + nums[h]
                if s == target:
                    break
                h+=1
            if s == target:
                break
            l+=1

        return [l,h]
        